apps/src/music/MusicView.jsx

```diff
@@ -261,10 +263,17 @@ class MusicView extends React.Component {
         new Blockly.FieldDropdown(function() {
           var options = [['anything', 'anything']];
           if (self.state.samplePanel && self.state.samplePanel !== 'main') {
-            const sounds = self.getCurrentSamplePackSounds();
-            options = sounds.map(sound => {
-              return [sound.id, sound.id];
-            });
+            const folders = self.getCurrentSamplePackSounds();
+            options = folders
+              .map(folder => {
+                return folder.sounds.map(sound => {
+                  return [
+                    folder.name + '/' + sound.id,
```

Comment:
` folder.name + '/' + sound.id `
This should be a function since it is shared across different parts of the code.

Reply:
Also a bit confusing why it has to be added twice from just reading the code and without context

Reply:
Thanks for looking at this.  Just to set some context: this is purely a prototype and code quality is not ship quality.  I'm merging these PRs into the `music-prototype` branch without reviews, but just made a pull request to break up the work.

Feel free to spectate, and I'll definitely let the org know if we go forward with this at a higher quality bar!